.github/workflows/build-and-test.yaml

```
@@ -2,27 +2,44 @@ name: build-and-test
 
 on:
   push:
-  schedule:
-    - cron: 0 0 * * *
+    branches:
+      - main
   workflow_dispatch:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  CC: /usr/lib/ccache/gcc
+  CXX: /usr/lib/ccache/g++
+
 jobs:
   build-and-test:
-    if: ${{ github.event_name != 'push' || github.ref_name == github.event.repository.default_branch }}
-    runs-on: ubuntu-latest
-    container: ${{ matrix.container }}
+    runs-on: codebuild-autoware-us-east-1-${{ github.run_id }}-${{ github.run_attempt }}-ubuntu-7.0-large
+    container: ${{ matrix.container }}${{ matrix.container-suffix }}
     strategy:
       fail-fast: false
       matrix:
         rosdistro:
           - humble
+        container-suffix:
+          - -cuda
         include:
           - rosdistro: humble
-            container: ros:humble
+            container: ghcr.io/autowarefoundation/autoware:latest-autoware-universe
```

```suggestion
            container: ghcr.io/autowarefoundation/autoware:universe-devel
```

Should this be this container instead? As in the current universe.

https://github.com/autowarefoundation/autoware.universe/blob/e0002f3e36defd695a026df741b5761df615d7e1/.github/workflows/build-and-test.yaml#L30C24-L30C74

---

Thanks. I updated the container image.